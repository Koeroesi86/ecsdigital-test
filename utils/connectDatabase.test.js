const sqlite3 = require('sqlite3');
const path = require('path');
const connectDatabase = require('./connectDatabase');

jest.mock('sqlite3', () => ({
  Database: jest.fn((p, callback) => {
    setTimeout(() => callback(null), 0);
    return {};
  }),
}));

describe('The connectDatabase', () => {
  beforeEach(() => {
    connectDatabase.reset();
  });

  it('should call sqlite3', async () => {
    const testPath = path.resolve('example');
    await connectDatabase(testPath);

    expect(sqlite3.Database.mock.calls).toMatchSnapshot();
  });

  it('should reuse database instance', async () => {
    const testPath = path.resolve('example');

    await connectDatabase(testPath);
    await connectDatabase(testPath);

    expect(sqlite3.Database.mock.calls).toMatchSnapshot();
  });

  it('should handle failure', async () => {
    const testPath = path.resolve('example');
    sqlite3.Database.mockImplementationOnce((p, callback) => callback(new Error('test')));

    try {
      await connectDatabase(testPath);
    } catch (e) {
      return expect(e).toMatchSnapshot();
    }

    expect(1).toEqual(2);
  });
});

