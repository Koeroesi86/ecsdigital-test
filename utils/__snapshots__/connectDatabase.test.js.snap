// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The connectDatabase should call sqlite3 1`] = `
Array [
  Array [
    "D:\\\\Chris\\\\Documents\\\\Developement\\\\tests\\\\ecsdigital\\\\ecsdigital-test\\\\example",
    [Function],
  ],
]
`;

exports[`The connectDatabase should handle failure 1`] = `[Error: test]`;

exports[`The connectDatabase should reuse database instance 1`] = `
Array [
  Array [
    "D:\\\\Chris\\\\Documents\\\\Developement\\\\tests\\\\ecsdigital\\\\ecsdigital-test\\\\example",
    [Function],
  ],
  Array [
    "D:\\\\Chris\\\\Documents\\\\Developement\\\\tests\\\\ecsdigital\\\\ecsdigital-test\\\\example",
    [Function],
  ],
]
`;
